mergify_engine/queue/merge_train.py

```diff
@@ -1154,6 +1149,20 @@ async def _split_failed_batches(self, queue_rules: rules.QueueRules) -> None:
                 self.log.info(
                     "spliting failed car", position=current_queue_position, car=car
                 )
+
+                queue_name = car.still_queued_embarked_pulls[0].config["name"]
+                try:
+                    queue_rule = queue_rules[queue_name]
+                except KeyError:
+                    # We just need to wait the pull request has been removed from
+                    # the queue by the action
+                    self.log.info(
+                        "can split failed batch TrainCar, queue rule does not exists anymore",
```

Comment:
```suggestion
                        "cant split failed batch TrainCar, queue rule does not exist anymore",
```
not sure it's really clear though 😢